### ISSUE TITLE
Add support for subdirectories when adding git repos as a library

### ISSUE THREAD

**Issue by jimbobbennett** [platformio/platformio-core#3990]
Some libraries are available as folders inside a Git repo. One example being ArduCAM.

The Arduino code for the ArduCAM is in this repo: https://github.com/ArduCAM/Arduino. Rather than have one library in this repo, it contains multiple libraries each inside a folder in this repo. It would be a great enhancement to be able to specify a folder inside a git repo for a library dependency to remove the need to configure library directories, or download the repo and copy one folder into the `lib` folder.

For example, being able to write something like:

```ini
lib_deps =
    https://github.com/ArduCAM/Arduino/tree/master/ArduCAM
```

**Comment by TheButlah** [comment]
This would also help pio work better with monorepos!

**Comment by DreamPWJ** [comment]
This feature is important

**Comment by christhomas** [comment]
Yeah this would be a very useful feature

**Comment by dmezh** [comment]
Agreed, would be quite useful

**Comment by teejaydub** [comment]
It would also make it possible to override individual platform libraries with forks, which unfortunately I need to do, and the workaround (separate management of a local copy of that repo along with a symlink:// lib_dep) is error-prone.

**Comment by nicklasb** [comment]
I would agree to this, a whole bunch of ESP-IDF components, for example, are in the [ESP-BSP](https://github.com/espressif/esp-bsp/tree/master/components)-repository and I cannot add them as it is now.
There are many more examples of this, please add this functionality. 

Accepting a https://github.com/espressif/esp-bsp.git/components/esp_lvgl_port-syntax, by identifying the break at the ".git", or using some other scheme. Or add a "path", as in idf_component.yml. It is quite usable. 
Maybe the best variant, as then you could combine with other schemes, and it is the most clear.

**Comment by nicklasb** [comment]
I also just noted that the maximum length of the URL, it seems like it is only 100 characters, which is super tight and will probably run out in many cases. The max length for most browsers are around 2000 characters, and several permanent link schemes has more than 100 characters. 
Basically I see no 2024 reason for restricting the URL length in such a severe manner.
